### PR TITLE
Add a deprecation notice for `localStorageClassName` field at HyperConvergedSpec

### PR DESCRIPTION
**Pull request by orenc1**
This field has currently no effect.
Adding a deprecation notice since we can't remove it to preserve backward compatibility.

Also, fixing several typos in some docs and docstrings.

Signed-off-by: orenc1 <ocohen@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
deprecate localStorageClassName
```

### CODE REVIEW

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type HyperConvergedSpec struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	// LocalStorageClassName the name of the local storage class.
+	// **[DEPRECATED]** LocalStorageClassName the name of the local storage class.
```
Comment (nunnatsa) [inline api/v1beta1/hyperconverged_types.go:37]: While I couldn't find a kubebuilder marker for deprecation, golang does support deprecation of fields/function/types. Please add the golang deprecation:

```suggestion
	// **[DEPRECATED]** LocalStorageClassName the name of the local storage class.
	//
	// deprecated
```

Comment (orenc1) [reply api/v1beta1/hyperconverged_types.go:37]: Done.
It indeed marks the field with strike-through (at least in GoLand IDE).

Comment (nunnatsa) [reply api/v1beta1/hyperconverged_types.go:37]: After reading around, there is a bettergolang format for that:
```suggestion
	// Deprecated: LocalStorageClassName the name of the local storage class.
```

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**